src/main/java/com/palantir/ri/ResourceIdentifier.java

```diff
@@ -297,6 +298,38 @@ public static ResourceIdentifier of(
         return of(service, instance, type, builder.toString());
```

Comment:
I wonder how common this constructor is in practice, the StringBuilder could be trivially presized to match the final locator length. The new parsing methods should all be able to operate on CharSequence rather than String, so the builder could be passed through without `toString`. I'm not sure if jep280 optimizations special-case CharSequence implementations without calling toString though, so that may not buy us anything.

Reply:
Yeah this seems like a good idea. Will save this for a future PR.